### PR TITLE
add doctests to TF ViT

### PR DESCRIPTION
**Pull request by johko**
# What does this PR do?

Add doctests for the TF version of ViT

Fixes # (issue)
#16292 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@patrickvonplaten

### CODE REVIEW

File: src/transformers/models/vit/modeling_tf_vit.py [hunk 1]
```diff
@@ -662,7 +662,6 @@ def call(
         Returns:
 
         Examples:
-
```
Comment (johko) [inline src/transformers/models/vit/modeling_tf_vit.py:665]: I wasn't sure if there is supposed to be an empty line here or not, for me it looked better without it

File: src/transformers/models/vit/modeling_tf_vit.py [hunk 2]
```diff
@@ -677,6 +676,8 @@ def call(
         >>> inputs = feature_extractor(images=image, return_tensors="tf")
         >>> outputs = model(**inputs)
         >>> last_hidden_states = outputs.last_hidden_state
+        >>> list(last_hidden_states.shape)
+        [1, 197, 768]
         ```"""
```
Comment (johko) [inline src/transformers/models/vit/modeling_tf_vit.py:681]: Also wasn't sure about the style here, keep ``` and """ in the same line or split to two lines

**Comment by ydshieh** [comment]
Hi, @johko

Thank you for this PR! Really appreciate.

I just realized that we have [PT_VISION_BASE_MODEL_SAMPLE](https://github.com/huggingface/transformers/blob/c85547af2b69f9082bcd7bac97092b1d162f3fdc/src/transformers/utils/doc.py#L541), but not `TF_VISION_BASE_MODEL_SAMPLE`.

Ideally, we would like to re-use code sample in [doc.py](https://github.com/huggingface/transformers/blob/main/src/transformers/utils/doc.py). I will need to discuss with the team to make a decision.

I will come back to you once we have a decision. Sorry for the inconvenience!

File: src/transformers/models/vit/modeling_tf_vit.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/models/vit/modeling_tf_vit.py:665]: Let's just keep the empty line. It is done this way in other places.

File: src/transformers/models/vit/modeling_tf_vit.py (hunk 2 above)
Comment (ydshieh) [reply src/transformers/models/vit/modeling_tf_vit.py:681]: It should be more or less a style choice. In the model files, we usually keep it as **```"""**

**Comment by ydshieh** [comment]
@johko 

```
TF_VISION_BASE_MODEL_SAMPLE
TF_VISION_SEQ_CLASS_SAMPLE
```
are added in this PR (and already merged to `main`) #16477

Once you pull the upstream's `main` in your local clone's `main` (or `master`), and rebase or merge it into your working branch, the process is just a matter of using **@add_code_sample_docstrings** together with **expected_output**, **checkpoint**, etc. You can take this as a reference:

https://github.com/huggingface/transformers/pull/16363/files#diff-5707805d290617078f996faf1138de197fa813f78c0aa5ea497e73b5228f1103

Regarding the checkpoint to use: 

- TFViTModel
  "google/vit-base-patch16-224-in21k"

- TFViTForImageClassification
  "google/vit-base-patch16-224"

I manually checked the code sample. Let me know if you encounter any issue.

**Comment by johko** [comment]
@ydshieh 
Thanks for the explanation, I'll look into it and implement the changes within the next days

**Comment by johko** [comment]
@ydshieh 
Ah, sorry I messed up the history now, I'm still getting used to the VS Code Git plugin.
But I added the `add_code_sample_docstrings` decorator to `modeling_tf_vit.py` now

**Comment by johko** [comment]
@ydshieh I'll close this PR and create a new one with a clean history

**Comment by ydshieh** [comment]
No problem, @johko . Thank you for the effort!

**Pull request closed without merging**